### DISCUSSION TITLE
qt 6.2 Monterey bottles

### DISCUSSION THREAD

**Discussion by viking1304** [Homebrew/discussions#2361] [Everyday usage] [answered]
Since qt 6.2 is not bottled for Monterey because compilation fails, I decided to download 6.2 and to try to compile it myself.

When I opened https://download.qt.io/official_releases/qt/6.2/ I noticed that there are two versions available for download 6.2.0 and 6.2.1. For some reason 6.2.1 only exists in that folder and there is absolutely no information about it anywhere. It is not announced anywhere for two days, and all links still points to 6.2.0, so I am not sure should this "phantom edition" of qt be used for formula or not.

I absolutely have no experience in updating formulas, so my experiment filed with a message that manifest cannot be found for 6.2.1. I would be grateful if someone could point me in the right direction, at least with reference links with articles that I should read.

Then I tried to compile 6.2.1 by hand. I applied all the patches that I found in formula, but I am not sure if I applied correct parameters to configure. It works for some time but then it fails.

I would be very grateful if someone can help me with right parameters that should be used for configure. I will try to compile both 6.2.0 and 6.2.1 and I will let you know about results, so maybe we can figure out together if formula can be fixed or there is an issue that should be reported upstream.

**Comment by carlocab** [comment, marked as answer]
Try updating the `url` line and the `sha256` line (you can use `curl -L <tarball url> | shasum -a 256`), and then do `brew install --build-from-source qt` to attempt a build according to the specifications in the formula. You need the `--build-from-source` flag to prevent `brew` from looking for a non-existent bottle manifest.

**Reply by viking1304** [reply]
Thank you Carlo!

I just found official announcement of 6.2.1, so it's not a phantom edition of qt any more :)
https://www.qt.io/blog/qt-6.2.1-released

I tried both 6.2.0 and 6.2.1 and they both have exactly the same problem. 

6.2.0

```
/tmp/qt-20211027-71565-1er2hg9/qt-everywhere-src-6.2.0/qtbase/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h:56:32: error: unknown type name 'CGColorSpaceRef'; did you mean 'QColorSpace'?
    void setColorSpace(QCFType<CGColorSpaceRef> colorSpace);
    
/tmp/qt-20211027-71565-1er2hg9/qt-everywhere-src-6.2.0/qtbase/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.mm:90:32: error: out-of-line definition of 'setColorSpace' does not match any declaration in 'QIOSurfaceGraphicsBuffer'
void QIOSurfaceGraphicsBuffer::setColorSpace(QCFType<CGColorSpaceRef> colorSpace)

2 errors generated.
```

6.2.1

```
/tmp/qt-20211027-84784-57ou81/qt-everywhere-src-6.2.1/qtbase/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h:56:32: error: unknown type name 'CGColorSpaceRef'; did you mean 'QColorSpace'?
    void setColorSpace(QCFType<CGColorSpaceRef> colorSpace);
 
/tmp/qt-20211027-84784-57ou81/qt-everywhere-src-6.2.1/qtbase/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.mm:90:32: error: out-of-line definition of 'setColorSpace' does not match any declaration in 'QIOSurfaceGraphicsBuffer'
void QIOSurfaceGraphicsBuffer::setColorSpace(QCFType<CGColorSpaceRef> colorSpace)

2 errors generated.
```

Unfortunately I have no idea why this fails and can't find any info about it. I just hope that someone else does.

**Reply by carlocab** [reply]
You could try checking the Qt issue tracker. Someone's bound to have reported this by now. If not, perhaps you would consider doing so.

**Comment by viking1304** [comment]
I as I already wrote in [this pull request comment](https://github.com/Homebrew/homebrew-core/pull/88168#issuecomment-953466312), I have managed to successfully compile qt 6.2.1 on M1, but only with full XCode. 

I am totally confused how 6.2.0 was even built and bottled, since `qtwebengine` can not be compiled with `CommandLineTools` and there is a missing include in `qtbase/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h`. 😱 😱 😱 

Worst of all is that `qtwebengine` compilation requires `XCode` and `python 2.7`! 😱 

@carlocab if you think that more detailed conversation could be useful in pull request comments, we can close this discussion and continue there.

**Reply by carlocab** [reply]
Sure, we can continue the discussion in the PR. CC @paperchalice for feedback.

Thanks so much for investigating this!

**Comment by tryittogether52** [comment]
It does not only occurs on Monterey, it also occurs on Big sur (macOS 11.6 ). How to fix?

Read this: https://code.qt.io/cgit/qt/qtbase.git/commit/src/plugins/platforms/cocoa?id=dece6f5840463ae2ddf927d65eb1b3680e34a547
